src/ir/auth_logic/souffle_emitter.h

```diff
@@ -102,14 +120,48 @@ class SouffleEmitter {
     return absl::StrCat(".decl ", predicate.name(), "(", arguments, ")");
   }
 
-  std::string EmitDeclarations() {
-    std::vector elements_vec(std::make_move_iterator(declarations_.begin()),
-                             std::make_move_iterator(declarations_.end()));
-    std::sort(elements_vec.begin(), elements_vec.end());
-    return absl::StrJoin(elements_vec, "\n",
-                         [this](std::string* out, auto decl) {
-                           return absl::StrAppend(out, EmitDeclaration(decl));
-                         });
+  std::string EmitArguments(std::vector<datalog::Argument> arguments) {
```

Comment:
const reference instead of value here: `const vector...& arguments`?

Reply:
Oops. Missed this!